### PR TITLE
libtiff upgrade to version 4.6.0

### PR DESCRIPTION
**Pull request by klatism**
Upgrade libtiff dependency from version 4.2.0 to 4.6.0.

- [X] I agree to contribute to the project under Apache 2 License.
- [X] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [X] The PR is proposed to the proper branch
- [ ] There is a reference to the original bug report and related work
- [ ] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by asmorkalov** [comment]
Looks like `libport.h` and `snprintf.c` are not present in libtiff 4.6.0 and may be removed from OpenCV too.

**Comment by asmorkalov** [comment]
@opencv-alalek should we update CMake with the original project CMake as they diverge significantly.

**Comment by opencv-alalek** [comment]
Please ignore CMake warnings - they are expected (internal checks of build configurations).
Please take a look on compilation warnings (e.g., on Windows, Mac).

**Review by asmorkalov — APPROVED** [review]
:+1: Looks good to me. I propose to drop versioninfo.rc, makefiles and other files that are not used in OpenCV build.

**Comment by klatism** [comment]
Thanks! I'll try to get rid of those unused files
Also, looking into that failing Android test. Not sure yet what happened there. May need some help.

**Comment by opencv-alalek** [comment]
> failing Android test

Please ignore. It is CI infrastructure problem.

**Comment by asmorkalov** [comment]
@opencv-alalek Any sugesstions? May I merge the pr?

**Comment by klatism** [comment]
removed more files yesterday, which seems to have a peculiar effect - Android CI is now green, but Ubuntu started to fail. I'm wondering whether it may be an infrastructure problem as well.

**Comment by opencv-alalek** [comment]
Thank you for update!

Ignore, these CI builders fails on other PRs too in the same way (not related to this PR).

File: CMakeLists.txt [hunk 1]
```diff
@@ -227,7 +227,7 @@ endif()
 OCV_OPTION(OPENCV_ENABLE_NONFREE "Enable non-free algorithms" OFF)
 
 # 3rd party libs
-OCV_OPTION(OPENCV_FORCE_3RDPARTY_BUILD   "Force using 3rdparty code from source" OFF)
+OCV_OPTION(OPENCV_FORCE_3RDPARTY_BUILD   "Force using 3rdparty code from source" ON)
```
Comment (opencv-alalek) [inline CMakeLists.txt:230]: Please revert this "validation" change before merge.

**Review by opencv-alalek — APPROVED** [review]
Well done! Thank you for contribution :+1:

**Comment by klatism** [comment]
the temporary change to OPENCV_FORCE_3RDPARTY_BUILD is reverted. Ready to submit.

**Comment by klatism** [comment]
@asmorkalov @opencv-alalek, just checking if there's anything else left to do in this PR, or it can be merged.

**Pull request merged**